Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py

```diff
@@ -894,6 +943,152 @@ def reputation_operation_command(client_obj, indicator, reputation_function):
         return_outputs(*reputation_function(client_obj, artifact))
 
 
+def group_infos_by_alert_asset_name(asset_alerts):
+    """
+    this method converts assets with multiple alerts into assets per asset_alert and
+        returns both human readable and context.
+    For an asset, group the asset_alert infos based on asset_alert name.
+    Returns human readable and context data.
+
+    :param asset_alerts: normalized asset alerts returned by Backstory.
+    :return: both human readable and context format having asset per alerts object
+    """
+
+    unique_asset_alerts_hr = {}  # type: Dict[str,Any]
+    unique_asset_alert_ctx = {}  # type: Dict[str,Any]
+    for asset_alert in asset_alerts:
+        for info in asset_alert['AlertInfo']:
+            asset_alert_key = asset_alert['AssetName'] + '-' + info['Name']
+
+            asset_alert_hr = unique_asset_alerts_hr.get(asset_alert_key, {})
+            asset_alert_ctx = unique_asset_alert_ctx.get(asset_alert_key, {})
+
+            if asset_alert_hr:
+                # Re calculate First and Last seen time
+                if info['Timestamp'] >= asset_alert_hr['Last Seen Ago']:
+                    asset_alert_hr['Last Seen Ago'] = info['Timestamp']
+                    asset_alert_hr['Last Seen'] = get_informal_time(info['Timestamp'])
+                    asset_alert_ctx['LastSeen'] = info['Timestamp']
+                elif info['Timestamp'] <= asset_alert_hr['First Seen Ago']:
```

Comment:
i think this one might needs to be `if` not elif

Reply:
no @ronykoz , we wanted to calculate First and Last seen time and this condition works fine.